plugin/trino-hive/src/main/java/io/trino/plugin/hive/SortingFileWriter.java

```diff
@@ -227,10 +229,8 @@ private void mergeFiles(Iterable<TempFile> files, Consumer<Page> consumer)
                     .forEachRemaining(consumer);
 
             for (TempFile tempFile : files) {
-                Path file = tempFile.getPath();
-                if (!fileSystem.delete(file, false)) {
-                    throw new IOException("Failed to delete temporary file: " + file);
-                }
+                String file = tempFile.getPath();
+                fileSystem.deleteFile(file);
```

Comment:
Super nit: I would just inline this
```suggestion
                fileSystem.deleteFile(tempFile.getPath());
```